src/utils/updateDBIcons.ts

```diff
@@ -0,0 +1,129 @@
+import axios from 'axios'
+import util from 'util'
+import cmd from 'node-cmd'
+import { iconsServices, IconInterface, IIconsModel } from 'components/icons'
+import { infoServices } from 'components/info'
+import { Logger, NodeLogger } from 'helpers'
+import { getEncodedLink, isNewIcon } from './tools'
+import prepareUpdatedIcon from './tools/updatingIcons/prepareTheUpdate'
+import { getFilled, getOutlined } from './tools/updatingIcons/getSvgCode'
+import configs from 'configs'
+// promise-style version of cmd.get
+const pcmdGet = util.promisify(cmd.get)
+
+const updateIconsLogger = new Logger('updateDBIcons')
+
+const updateDBIcons = async (notifiedByHook = false) => {
+  try {
+    // Latest Database version's info:
+    const currentVersionInfo = await infoServices.getLatestVersionInfo()
+    // in initial setup, there wouldn't be any versions stored in the db, that's why we set an old date.
+    const currVersionDate = new Date(currentVersionInfo ? currentVersionInfo.createdAt : '1999-03-20')
+    if (!notifiedByHook) {
+      // if not notified by hook, we need to make sure that there's a new release first
+      const eosIconsTAGS = await axios.get(configs.GITLAB_TAGS_API)
+      const latestReleaseDate = eosIconsTAGS.data[0].commit.committed_date
+      if (latestReleaseDate <= currVersionDate) {
+        updateIconsLogger.logInfo('', { message: 'icons is up to date, updating process is finished' })
+        return
+      }
+    }
```

Comment:
In most cases the update icons process will be called without being notified by the gitlab hook (will be created in the next PR), so we need to make sure that there's a new version of icons before starting the process, if a hook notified our backend that there's a new release, this check will not be made.